src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -126,15 +127,17 @@ public function decode($data, $format, array $context = array())
                 $nbHeaders = $nbCols;
 
                 foreach ($cols as $col) {
-                    $headers[] = explode($keySeparator, $col);
+                    $headerChain = explode($keySeparator, $col);
```

Comment:
Is it necessary to create this intermediary variable? Could `$headerChain` disappear and just use `$headers` as before?

Reply:
```
$headers[] = explode($keySeparator, $col);
$headerCount[] = count(end($headers));
```
This would be ok?

Reply:
end() creates a reference - that's ok on 7.0, but slow on 5.6

Reply:
Yep, tested it on 7.1. I don't think it's a good idea to make fix that work on 7.0+ only. So I better stay with intermediary var. Except you prefer code like this
`$headerCount[] = count($headers[] = explode($keySeparator, $col));`